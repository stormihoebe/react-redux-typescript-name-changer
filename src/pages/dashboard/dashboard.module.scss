@import "../../assets/colors.scss";

.max_width {
    max-width: 1260px;
}

.dashboard {
    display: flex;
    flex: 1 0 100%;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

