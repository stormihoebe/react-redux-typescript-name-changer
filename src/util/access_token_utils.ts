export const isLoggedIn = (key: string  = "access-token"): boolean => {
    const result: string | null = localStorage.getItem(key)
    return result !== null
}

export const getAccessToken = (): string => {
    const accessToken = localStorage.getItem("access-token")
    return accessToken ? accessToken : ""
}

export const deleteAccessToken = (): void => {
    localStorage.removeItem("access-token")
}

export const setAccessToken = (token: string): void => {
     localStorage.setItem("access-token", token)
}
