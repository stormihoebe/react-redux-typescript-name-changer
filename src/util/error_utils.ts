export const parseError = (statusCode: number) => {
    switch (statusCode) {
            case 403:
                return "Username and Password not found"
            case 401:
                return "You are not logged in"
            default:
                return "A server error occurred"
        }
}