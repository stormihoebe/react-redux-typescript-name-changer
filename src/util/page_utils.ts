import { RouteComponentProps } from "react-router"

export const genericRouteProps = {} as RouteComponentProps<{}>

export function setPageTitle(title: string): void {
    document.title = `${title} | Your App`
}