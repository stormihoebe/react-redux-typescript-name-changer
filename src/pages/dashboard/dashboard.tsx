import * as React from "react"

import { RouteComponentProps } from "react-router"

import * as redux from "redux"
import { connect } from "react-redux"
import { AppState, Session } from "../../redux/core"
import { setPageTitle } from "../../util/page_utils"
import { Navbar } from "../../components/navbar/navbar"
import { Button } from "../../components/button/button"
import * as actions from "../../redux/actions"

const styles = require("./dashboard.module.css")

interface ReduxState {
    readonly session: Session,
}

interface ReduxActions {
    readonly logout: () => void
    readonly bootstrap: () => void
}

interface DashboardProps extends RouteComponentProps<{}> {
}

const mapStateToProps = (state: AppState, ownProps: DashboardProps): ReduxState => ({
    session: state.session,
})

const mapDispatchToProps = (dispatch: redux.Dispatch<AppState>): ReduxActions => ({
    logout: () => dispatch( actions.logout() ),
    bootstrap: () => dispatch( actions.bootstrap() ),
})

interface DashboardState {
}

export class Dashboard extends React.Component<ReduxState & ReduxActions & DashboardProps, DashboardState> {
    constructor(props: ReduxState & ReduxActions & DashboardProps) {
        super(props)

        this.state = {
        }
    }

    componentWillMount(): void {
        setPageTitle("Dashboard")
        this.props.bootstrap()
    }

    goToLogin = () => this.props.history.push("/")

    logout = () => {
        this.props.logout()
        this.goToLogin()
    }

    render(): JSX.Element {
        return (
            <div className={styles.dashboard}>
                <Navbar
                    logout={this.logout}
                    userName="Some User (click here to logout)"
                />
                <br />
                <br />

                <Button >
                    Hi there
                </Button>
            </div>
        )
    }
}

const ConnectedApp: React.ComponentClass<DashboardProps> = connect(mapStateToProps, mapDispatchToProps)(Dashboard)
export default ConnectedApp