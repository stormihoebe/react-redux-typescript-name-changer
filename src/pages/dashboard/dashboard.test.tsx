import * as React from "react"
import * as ReactDOM from "react-dom"
import { Dashboard } from "./dashboard"
import { create as renderer } from "react-test-renderer"
import { genericRouteProps } from "../../util/page_utils"

const actions = {
    logout: () => ({}),
    bootstrap: () => ({}),
    session: {
       loggedIn: false,
       accessToken: "",
    },
}

const render = () => (
    <Dashboard
        {...genericRouteProps}
        logout={actions.logout}
        bootstrap={actions.bootstrap}
        session={actions.session}
    />
)

it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(render(), div)
})

it("renders a snapshot", () => {
    const tree = renderer(render()).toJSON()
    expect(tree).toMatchSnapshot()
})