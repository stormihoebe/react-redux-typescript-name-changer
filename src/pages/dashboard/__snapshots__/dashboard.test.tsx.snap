exports[`test renders a snapshot 1`] = `
<div
  className="dashboard">
  <nav
    className="pt-navbar .modifier nav_bar">
    <div
      className="container">
      <div
        className="pt-navbar-group pt-align-right">
        <button
          className="pt-button pt-minimal user_menu">
          <span
            className="user_name"
            onClick={[Function]}>
            Some User (click here to logout)
          </span>
          <span
            className="pt-icon-standard pt-icon-chevron-down user_menu_chevron" />
        </button>
      </div>
    </div>
  </nav>
  <br />
  <br />
  <button
    className="pt-button button"
    disabled={undefined}
    onClick={undefined}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button">
    <span>
      Hi there
    </span>
  </button>
</div>
`;
